src/alerts/flash-refactor.scss

```diff
@@ -0,0 +1,133 @@
+  padding: $spacer-2;
+  color: var(--color-fg-default);
+  background-image: linear-gradient(var(--color-accent-subtle), var(--color-accent-subtle));
+  border: $border-width $border-style var(--color-accent-muted);
+  border-radius: $border-radius;
+  grid-auto-flow: column;
+  grid-template-areas: 'visual message actions close';
+  grid-template-columns: min-content 1fr min-content min-content;
+  grid-template-rows: min-content;
+
+  // `sm` breakpoint variantion
+  @media (max-width: #{map-get($breakpoints, 'sm') - 0.02px}) {
+    grid-template-areas:
+      'visual message close'
+      '  .    actions actions';
+    grid-template-columns: min-content 1fr min-content;
+    grid-template-rows: min-content min-content;
+
+    .Flash-actions {
+      margin: $spacer-2 0 0 $spacer-2;
+    }
+  }
+
+  // Elements
+
+  .Flash-visual {
+    display: grid;
+    // stylelint-disable-next-line primer/spacing
+    padding: calc((#{$spacer-5} - 20px) / 2) $spacer-2;
+    grid-area: visual;
+    align-self: start;
+
+    > .octicon {
+      margin-block: calc((#{$spacer-1}) / 2);
+    }
+
+    > * {
+      align-self: center;
+    }
+  }
+
+  .Flash-message {
+    // stylelint-disable-next-line primer/spacing
+    padding: calc((#{$spacer-5} - 20px) / 2) $spacer-2;
+    grid-area: message;
+    align-self: center;
+
+    p:last-child {
+      margin-bottom: 0;
+    }
+  }
+
+  .Flash-actions {
+    grid-area: actions;
+
+    &:last-child {
+      align-self: end;
+    }
+  }
+
+  .Flash-close {
+    grid-area: close;
+    margin: 0 0 0 $spacer-2;
+
+    button {
+      &:hover {
+        background-color: var(--color-action-list-item-default-hover-bg);
+      }
+    }
+  }
+
+  .Flash-visual .octicon {
+    color: var(--color-accent-fg);
+  }
+
+  &.Flash--warning {
+    color: var(--color-fg-default);
+    background-image: linear-gradient(var(--color-attention-subtle), var(--color-attention-subtle));
+    border-color: var(--color-attention-muted);
+
+    .Flash-visual .octicon {
+      color: var(--color-attention-fg);
+    }
+  }
+
+  &.Flash--error {
+    color: var(--color-fg-default);
+    background-image: linear-gradient(var(--color-danger-subtle), var(--color-danger-subtle));
+    border-color: var(--color-danger-muted);
+
+    .Flash-visual .octicon {
+      color: var(--color-danger-fg);
+    }
+  }
+
+  &.Flash--success {
+    color: var(--color-fg-default);
+    background-image: linear-gradient(var(--color-success-subtle), var(--color-success-subtle));
+    border-color: var(--color-success-muted);
+
+    .Flash-visual .octicon {
+      color: var(--color-success-fg);
+    }
+  }
+
+  // Full-width
+
+  &.Flash--full {
+    // stylelint-disable-next-line primer/spacing
+    margin-top: ($border-width * -1);
+    border-right: 0;
+    border-left: 0;
+    border-radius: 0;
+  }
+
+  // Replace with Primitives' Narrow viewport range
+  @media (max-width: #{map-get($breakpoints, 'md') - 0.02px}) {
+    &.Flash--full-whenNarrow {
```

Comment:
`full-whenNarrow` 👍  This looks useful.

Reply:
@simurai I'm already using it for https://github.com/primer/css/pull/1898 ([Example in Storybook](https://primer-css-ejwtjxgp9-primer.vercel.app/css/storybook?path=/story/components-applicationframe-examples--repo-settings)) 🎉  🎉

Reply:
Yesss... this, exactly!